apihub/provision-apihub.sh

```diff
@@ -22,7 +22,18 @@
 # For more about the Registry Provisioning API that is used in this
 # script, see https://cloud.google.com/apigee/docs/api-hub/provision.
 
-# Usage: provision-apihub.sh -p GCP-PROJECT-ID -l us-central1 -u GCP-USER-EMAIL -r KEY-RING-NAME -k KEY-NAME
+# Prerequisites:
+# - A Google Cloud account (USER_EMAIL) with billing enabled.
+# - A Google Cloud project (PROJECT_ID).
+# - The gcloud tool (https://cloud.google.com/sdk/gcloud).
+# - A chosen location (RUNTIME_LOCATION) for the instance. Currently "us-central1"
+#   is recommended. Other supported locations are listed at
+#     https://cloud.google.com/apigee/docs/api-hub/provision
+# - A chosen CMEK (https://cloud.google.com/kms/docs/cmek) key ring name
+#   (KEY_RING_NAME) and key name (KEY_NAME).
+
+# Usage:
+#  provision-apihub.sh -u USER_EMAIL -p PROJECT_ID -l RUNTIME_LOCATION -r KEY_RING_NAME -k KEY_NAME
```

Comment:
This is more of a suggestion to make things simpler.

Can we make the CMEK optional? In GCP we can either provide a CMEK key or bypass and use Google managed CMEK. 

For Trial users, this can really speed up the provisioning since now they don't have to go through the step of provisioning a new CMEK. [Obviously we can prompt the user that this setting cannot be changed in the future]

Reply:
Interesting.. Can you provide an example of a GCP service that supports this? As far as I am aware, based on our PRD, it's a requirement for our customers to provide the CMEK during provisioning.

I had the impression that to be CMEK horizontal compliant the encryption key has to be customer managed.

Reply:
If you look at the CloudSQL configuration, CMEK is not required (but it is supported).

![Screen Shot 2022-05-19 at 6 19 35 PM](https://user-images.githubusercontent.com/6208023/169414222-563df5c8-6298-4183-87cf-e86f1a86d2e5.png)

Reply:
I see, thanks! We can follow up on this discussion internally.